### ISSUE TITLE
owned lookupfiles can't be edited

### ISSUE THREAD

**Issue by Muryoutaisuu** [LukeMurphey/lookup-editor#5]
Hi Luke

First of all: Thanks for your great app, it's used very broadly at our company!

We're having some issues with permissions/authentication.

Steps to reproduce:
1. UserA creates a lookuptable via upload over settings -> lookups -> Add new Lookup table file. That lookupfile will automatically have set UserA as the owner.
2. UserA shares the lookup file in the app for other users.
3. UserB (who is absolutely allowed to read/write the lookupfile) opens the Lookup Editor App and clicks on the by UserA newly created lookupfile to edit it

UserB will get following message in WebGUI: 
`<yellow exlamation mark icon> This lookup could not be loaded from server`

In the lookupeditor logfile I see following:

```
2016-10-19 11:41:26,248 INFO Retrieving lookup contents, namespace=search, lookup=myawesomelookup.csv, type=csv, owner=fiorettin, version=None
2016-10-19 11:41:26,257 ERROR Lookup file could not be loaded
Traceback (most recent call last):
  File "<string>", line 759, in get_lookup_contents
  File "<string>", line 715, in get_lookup
  File "<string>", line 584, in resolve_lookup_filename
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 548, in get
    return SplunkRESTManager(cls, sessionKey=sessionKey).get(id)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 528, in get
    entity = self._get_entity(id, host_path=host_path)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 444, in _get_entity
    return self._fix_entity(splunk.entity.getEntity(self.model.resource, None, sessionKey=self.sessionKey, uri=id))
  File "/opt/splunk/lib/python2.7/site-packages/splunk/entity.py", line 249, in getEntity
    serverResponse, serverContent = rest.simpleRequest(uri, getargs=kwargs, sessionKey=sessionKey, raiseAllErrors=True)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/rest/__init__.py", line 516, in simpleRequest
    raise splunk.AuthorizationFailed(extendedMessages=uri)
AuthorizationFailed: [HTTP 403] Client is not authorized to perform requested action; https://<host>:<port>/servicesNS/fiorettin/search/data/lookup-table-files/myawesomelookup.csv

```

When looking into `appserver/controllers/lookup_edit.py`, I see following line:

```
def get_lookup_contents(self, lookup_file, namespace="lookup_editor", owner=None, header_only=False, version=None, lookup_type=None, **kwargs):
```

I do not exactly know, whether on call of this, and also of other functions, the owner is correctly committed. When manually changing the owner to "nobody", everything works fine.

Could you please investigate?

Thanks!

Kind regards
Muryoutaisuu

**Comment by LukeMurphey** [comment]
Thanks for the clear repro steps; you steps were very clear.

Could you clarify if the user created the lookup as a "user-only" lookup in step 1? I can't repro it unless I select "user-only" in step 1. I just want to make sure I'm fixing the issue you are seeing.

**Comment by LukeMurphey** [comment]
I have a fix for this. I'm planning on releasing the fix in version 2.5.0.

**Comment by LukeMurphey** [comment]
I got a fix for this. Version 2.5.0 has a fix for this. This isn't the default yet so you will need to manually select it on Splunk-base. Alternatively, you can download it [here](http://lukemurphey.net/attachments/download/397/lookup_editor.tar.gz).

Let me know if that works for you.

**Comment by Muryoutaisuu** [comment]
Hi Luke

I've updated the Lookup Editor to version 2.5.0, but I'm still getting an error.
In this case, userA=fiorettin, userB=pzm:
```
2016-11-16 08:17:33,137 INFO Retrieving lookup contents, namespace=pzmapp, lookup=dc_availability_applications.csv, type=csv, owner=fiorettin, version=None
2016-11-16 08:17:33,147 ERROR Lookup file could not be loaded
Traceback (most recent call last):
  File "<string>", line 785, in get_lookup_contents
  File "<string>", line 741, in get_lookup
  File "<string>", line 589, in resolve_lookup_filename
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 548, in get
    return SplunkRESTManager(cls, sessionKey=sessionKey).get(id)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 528, in get
    entity = self._get_entity(id, host_path=host_path)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/models/base.py", line 444, in _get_entity
    return self._fix_entity(splunk.entity.getEntity(self.model.resource, None, sessionKey=self.sessionKey, uri=id))
  File "/opt/splunk/lib/python2.7/site-packages/splunk/entity.py", line 249, in getEntity
    serverResponse, serverContent = rest.simpleRequest(uri, getargs=kwargs, sessionKey=sessionKey, raiseAllErrors=True)
  File "/opt/splunk/lib/python2.7/site-packages/splunk/rest/__init__.py", line 516, in simpleRequest
    raise splunk.AuthorizationFailed(extendedMessages=uri)
AuthorizationFailed: [HTTP 403] Client is not authorized to perform requested action; https://<host>:8089/servicesNS/fiorettin/pzmapp/data/lookup-table-files/dc_availability_applications.csv
```

When manually calling ```https://<host>:8089/servicesNS/fiorettin/pzmapp/data/lookup-table-files/dc_availability_applications.csv``` via browser and successfully authenticating as UserB, I'm getting an XML file with:

```
<?xml version="1.0" encoding="UTF-8"?>
<response>
  <messages>
    <msg type="ERROR">You do not have permissions to access objects of user=fiorettin</msg>
  </messages>
</response>
```

When calling it as user pzm (userB) instead of fiorettin (userA) it works: ```https://<host>:8089/servicesNS/pzm/pzmapp/data/lookup-table-files/dc_availability_applications.csv``` So I think, you should change the value instead of the original owner to the user calling the lookupfile.

Kind regards
~ Muryoutaisuu

**Comment by LukeMurphey** [comment]
Can you check who is listed as the owner of the lookup file by running the following?

| rest /services/data/lookup-table-files count=0 | table title id "eai:acl.owner"

Version 2.5.0 uses the EAI owner field so I want to see if that matches what works.

BTW: you may want to make sure your browser cache is clear as this was a client-side fix and sometimes the Javascript doesn't get replaced.

**Comment by Muryoutaisuu** [comment]
Hi Luke

> BTW: you may want to make sure your browser cache is clear as this was a client-side fix and sometimes the Javascript doesn't get replaced.

This was the key...

I reloaded the Lookup Editor App in Chrome with DevTools open and Flag "Disable cache (while DevTools is open)" set and it worked...

Thanks for the fix and your patience, great work! :+1:

**Comment by rajanshrivastav** [comment]
Hi Luke,
I'm facing similar kind of issue where I'm not able to even edit the existing lookups. KV based lookup is working fine, but CSV lookup is not working.
I checked for permission issue and there is no permission issue.
I'm using 2.3.1 version. 
The lookup file is present on the server and has 775 permission to it.
Please help me resolving this issue.

Please let me know if you need any more info.

Thanks,

**Comment by LukeMurphey** [comment]
@rajanshrivastav: try upgrading to a newer version of the app. The fix mentioned in this ticket was fixed 2.5.0. You won't have the fix since you are on version 2.3.1.

**Comment by rajanshrivastav** [comment]
Thanks @LukeMurphey , It has been resolved now :+1:
